bundler/src/bundler/chunk/merge.rs

```diff
@@ -637,74 +637,46 @@ struct ImportMetaHandler<'a, 'b> {
-                                    Ident {
-                                        sym: js_word!("import"),
-                                        ..
-                                    },
-                                prop:
-                                    Ident {
-                                        sym: js_word!("meta"),
-                                        ..
-                                    },
-                                ..
-                            }) => match &*me.prop {
-                                Expr::Ident(Ident {
-                                    sym: js_word!("url"),
-                                    ..
-                                }) => {
-                                    let res = self.hook.get_import_meta_url(me.span, self.file);
-                                    match res {
-                                        Ok(v) => match v {
-                                            Some(expr) => {
-                                                *e = expr;
-                                                return;
-                                            }
-                                            None => {}
-                                        },
-                                        Err(err) => self.err = Some(err),
-                                    }
-                                }
-                                Expr::Ident(Ident {
-                                    sym: js_word!("main"),
-                                    ..
-                                }) if !self.is_entry => {
-                                    *e = Expr::Lit(Lit::Bool(Bool {
-                                        span: me.span,
-                                        value: false,
-                                    }));
-                                    return;
-                                }
-                                _ => {}
-                            },
-
-                            _ => {}
-                        },
-                    }
+            Expr::MetaProp(MetaPropExpr {
+                meta:
+                    Ident {
+                        sym: js_word!("import"),
+                        span,
+                        ..
+                    },
+                prop:
+                    Ident {
+                        sym: js_word!("meta"),
+                        ..
+                    },
+                ..
+            }) => match self.hook.get_import_meta_props(
+                *span,
+                &ModuleRecord {
+                    file_name: self.file.to_owned(),
+                    is_entry: self.is_entry,
+                },
+            ) {
+                // TODO(nayeemrmn): This substitutes any `import.meta` reference with a complete
+                // re-instantiated object. Support mutating `import.meta` using pre-declared a
+                // variable like `import_meta_1`.
```

Comment:
@kdy1 Regarding this, is there an easy way of getting a variable name not used as a local var in the module? If not I think it needs an extra pass, which I can also use to see if `import.meta` is ever used.

Reply:
You can use `swc_ecma_utils::private_ident!`.
`hygiene` pass will rename it to an uniqueidentifier.